src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -899,7 +905,9 @@ export const handleGetTemplateForm: ControllerHandler<{ formId: string }> = (
       // Step 2: Remove private form details before being returned.
       .map((populatedForm) => populatedForm.getPublicView())
       .map((scrubbedForm) =>
-        res.status(StatusCodes.OK).json({ form: scrubbedForm }),
+        res
+          .status(StatusCodes.OK)
+          .json({ form: scrubbedForm as PublicFormDto }),
```

Comment:
why are all the casts in this file necessary?

Reply:
mismatch of typings between server and client

for example, PublicFormView in the server has some _id typed as `ObjectId` which is not the same as  the`string` typing expected from the DTO. However, casting is allowed since `ObjectId` extends string (and can be serialized to the same thing).